tests/test_public_api.py

```diff
@@ -0,0 +1,58 @@
+"""Test myi public API."""
+
+import numpy as np
+
+from myia import myia, public_api
+
+
+def run_generator():
+    rstate = public_api.myia_random_initialize(None)
+    r0, v0 = public_api.myia_random_uniform(rstate, 0, 1, ())
+    r1, v1 = public_api.myia_random_uniform(r0, -5, -1, (2, 3))
+    r2, v2 = public_api.myia_random_uniform(r1, 10, 100, ())
+    r3, v3 = public_api.myia_random_uniform(r2, 50, 51, (3, 3, 2))
+    return r3, v0, v1, v2, v3
+
+
+@myia(backend='pytorch')
+def run_pytorch():
+    return run_generator()
+
+
+@myia(backend='relay')
+def run_relay():
+    return run_generator()
+
+
+def run_random_generator(fn):
+    r, v0, v1, v2, v3 = fn()
+    print(r)
+    print(v0)
+    print(v1)
+    print(v2)
+    print(v3)
+
+    assert 0 <= v0 < 1
+
+    assert v1.shape == (2, 3)
+    assert np.all(-5 <= v1)
+    assert np.all(v1 < -1)
+
+    assert 10 <= v2 < 100
+
+    assert v3.shape == (3, 3, 2)
+    assert np.all(50 <= v3)
+    assert np.all(v3 < 51)
+
+    return r
+
+
+def test_pytorch():
+    run_random_generator(run_pytorch)
+
+
+def test_relay():
+    rstate = run_random_generator(run_relay)
+    key, counter = rstate
+    assert key == 12345678
+    assert counter == 4
```

Comment:
At least for the Relay implementation, I would like to see an additional test that tests the exact values returned by the RNG with some specific seed. For example, initialize with seed 1234, generate one 2x2 matrix, check the values, generate a second 2x2 matrix, check the values. That way, if something is changed in the RNG implementation, we can verify that the same numbers come out.

Reply:
Ok!